examples/advanced_operations/add_responsive_search_ad_full.py

```diff
@@ -0,0 +1,535 @@
+    # apply GeoTargetConstant.resource_name.
+    gtc_request = client.get_type("SuggestGeoTargetConstantsRequest")
+    gtc_request._LOCALE = _LOCALE
+    gtc_request._COUNTRY_CODE = _COUNTRY_CODE
+
+    # The location names to get suggested geo target constants.
+    gtc_request.location_names.names.extend(
+        [_GEO_LOCATION_1, _GEO_LOCATION_2, _GEO_LOCATION_3]
+    )
+
+    results = geo_target_constant_service.suggest_geo_target_constants(gtc_request)
+
+    operations = []
+    for suggestion in results.geo_target_constant_suggestions:
+        geo_target_constant = suggestion.geo_target_constant
+        print(
+            f"{geo_target_constant.resource_name} "
+            f"({geo_target_constant.name}, "
+            f"{geo_target_constant._COUNTRY_CODE}, "
+            f"{geo_target_constant.target_type}, "
+            f"{geo_target_constant.status.name}) "
+            f"is found in _LOCALE ({suggestion._LOCALE}) "
+            f"with reach ({suggestion.reach}) "
+            f"from search term ({suggestion.search_term})."
+        )
+        # Create the campaign criterion for location targeting.
+        campaign_criterion_operation = client.get_type("CampaignCriterionOperation")
+        campaign_criterion = campaign_criterion_operation.create
+        campaign_criterion.campaign = campaign_resource_name
+        campaign_criterion.location.geo_target_constant = (
+            geo_target_constant.resource_name
+        )
+        operations.append(campaign_criterion_operation)
+
+    campaign_criterion_service = client.get_service("CampaignCriterionService")
+    campaign_criterion_response = campaign_criterion_service.mutate_campaign_criteria(
+        customer_id=customer_id, operations=[*operations]
+    )
+
+    for result in campaign_criterion_response.results:
+        print(f'Added campaign criterion "{result.resource_name}".')
+
+
+def add_images(client, customer_id, campaign_resource_name):
+    # Step 6.1 - Add Image Asset.
+
+    # Download PNG image from URL.
+    url = "https://gaagl.page.link/bjYi"
+    image_content = requests.get(url).content
+
+    asset_service = client.get_service("AssetService")
+    asset_operation = client.get_type("AssetOperation")
+    asset = asset_operation.create
+    asset.type_ = client.enums.AssetTypeEnum.IMAGE
+
+    # Data field is the raw bytes data of an image.
+    asset.image_asset.data = image_content
+    asset.image_asset.file_size = len(image_content)
+
+    # MIME type of the image (IMAGE_JPEG, IMAGE_PNG, etc.).
+    # See more types on the link below.
+    # https://developers.google.com/google-ads/api/reference/rpc/v11/MimeTypeEnum.MimeType
+    asset.image_asset.mime_type = client.enums.MimeTypeEnum.IMAGE_PNG
+    # Use your favorite image library to determine dimensions
+    asset.image_asset.full_size.height_pixels = 1200
+    asset.image_asset.full_size.width_pixels = 1200
+    asset.image_asset.full_size.url = url
```

Comment:
Is setting `file_size` and `full_size` necessary? We don't set any of those attributes in https://developers.google.com/google-ads/api/samples/upload-image#python or https://developers.google.com/google-ads/api/samples/upload-image-asset#python.

Reply:
I just tested by taking them out, and I didn't get any errors. Therefore, we could comment them out or remove them.